test/powershell/enginecore/CommandDiscovery.Tests.ps1

```diff
@@ -0,0 +1,105 @@
+﻿Describe "Command Discovery tests" -Tags "CI" {
+
+    BeforeAll {
+        setup -f testscript.ps1 -content "TestScript"
+		setup -f testscripp.ps1 -content "TestScripp"
+
+        $TestCasesCommandNotFound = @(
+                        @{command = 'CommandThatDoesnotExist' ; testName = 'Non-existent command'}
+                        @{command = 'testscrip?.ps1' ; testName = 'Multiple matches for filename'}
+                        @{command = "demo" + [System.IO.Path]::DirectorySeparatorChar; testName = 'Non existent command with directory separator'}
+                        @{command = [System.IO.Path]::DirectorySeparatorChar; testName = 'Directory separator'}
+                        @{command = 'environment::\path'; testName = 'Provider qualified path'}                        
+                       )
+    }
+        
+    It "<testName>" -TestCases $TestCasesCommandNotFound {
+        param($command)
+        try
+        {
+            & $command
+            throw "Command '$command' was found"
```

Comment:
As stated, the error text sounds like success. I think it could be clearer that it's a failure, e.g.: `Should not have found $command`